Chess/Board.cs

```diff
@@ -269,15 +269,14 @@ public bool IsKingUnderAttack(Color color)
     /// <returns></returns>
     private bool IsFieldUnderAttack(Vector field, Color color)
     {
-        var pieces = GetPieces(color);
-        foreach (var piece in pieces)
+        foreach (PieceType pieceType in Enum.GetValues(typeof(PieceType)))
         {
-            var possibleMoves = GetMoves(piece);
-            if (possibleMoves.Any(m => m.PieceNewPosition == field))
-            {
+            var pretendPiece = new Piece(pieceType, color.GetOppositeColor(), field);
```

Comment:
we could do `color.GetOppositeColor()` outside of the foreach loop for even faster performance